lib/containers/urls.pm

```diff
@@ -51,64 +52,229 @@ sub get_opensuse_registry_prefix {
+        },
+        '15.1' => {
+            released_registry => sub { "registry.opensuse.org/opensuse/leap:15.1" },
+            totest_registry   => sub {
+                if (check_var('ARCH', 'x86_64')) {
+                    "registry.opensuse.org/opensuse/leap/15.1/images/totest/containers/opensuse/leap:15.1";
+                } elsif (check_var('ARCH', 'aarch64') || check_var('ARCH', 'arm')) {
+                    "registry.opensuse.org/opensuse/leap/15.1/arm/images/totest/containers/opensuse/leap:15.1";
+                } else {
+                    "";
+                }
+            },
+            available_arch => ['x86_64', 'aarch64', 'arm']
+        },
+        '15.2' => {
+            released_registry => sub { "registry.opensuse.org/opensuse/leap:15.2" },
+            totest_registry   => sub {
+                if (check_var('ARCH', 'x86_64')) {
+                    "registry.opensuse.org/opensuse/leap/15.2/images/totest/containers/opensuse/leap:15.2";
+                } elsif (check_var('ARCH', 'aarch64') || check_var('ARCH', 'arm')) {
+                    "registry.opensuse.org/opensuse/leap/15.2/arm/images/totest/containers/opensuse/leap:15.2";
+                } else {
+                    "";
+                }
+            },
+            available_arch => ['x86_64', 'aarch64', 'arm']
+        },
+        '15.3' => {
+            released_registry => sub { "registry.opensuse.org/opensuse/leap:15.3" },
+            totest_registry   => sub {
+                if (check_var('ARCH', 'x86_64')) {
+                    "registry.opensuse.org/opensuse/leap/15.3/images/totest/containers/opensuse/leap:15.3";
+                } elsif (check_var('ARCH', 'aarch64') || check_var('ARCH', 'arm')) {
+                    "registry.opensuse.org/opensuse/leap/15.3/arm/images/totest/containers/opensuse/leap:15.3";
+                } else {
+                    "";
+                }
+            },
+            available_arch => ['x86_64', 'aarch64', 'arm']
+        }
+    }
+
+);
+
+sub check_image_arch_in_registry {
+    my ($distri, $version, $arch) = @_;
+    if (grep /\b$arch\b/, @{$registries{$distri}{$version}{available_arch}}) {
+        record_info "urls for $version@$arch", Dumper($registries{$distri}{$version});
+        return 1;
+    }
+    record_info "Supported archs", Dumper($registries{$distri}{$version}{available_arch}), result => 'fail';
+    return 0;
+}
+
 # Returns a tuple of image urls and their matching released "stable" counterpart.
 # If empty, no images available.
 sub get_suse_container_urls {
-    my $version    = shift // get_required_var('VERSION');
-    my $dotversion = $version =~ s/-SP/./r;                    # 15 -> 15, 15-SP1 -> 15.1
-    $dotversion = "${dotversion}.0" if $dotversion !~ /\./;    # 15 -> 15.0
+    my $version = shift // get_required_var('VERSION');
+    my $arch    = shift // get_required_var('ARCH');
+    my $distri  = shift // get_required_var('DISTRI');
 
     my @untested_images = ();
```

Comment:
`untested_images` as well as `released_images`  came from a previous team agreement. i dont want to interfere and change them. i think it is messy to change the names frequently. in the other side i think the structure's name as quite good and they do not mess with the already implementation.

Reply:
I like your `totest_` and `released_` prefixes. But I'd like to use the same prefixes everywhere.

Maybe another point: I'd rename `%registries` to `%images` and `released_registry` to `released_image` and `totest_registry` to `totest_image`.

Reply:
@jlausuch @Vogtinator can you help here so we agree on same convention and use it everywhere?

Reply:
Ping. I would still be for unifying those names. E.g. `released_images` and `totest_images` as we have above.

Reply:
Sorry for the late answer. 
I am in favor of renaming `%registries` to `%images`.

And now that I hear Pavel's opinion, I kind of like it. I am also in favor of using `totest_images` instead of `unreleased_images`, but @b10n1k if you don't want to do that change here, I'm fine with doing that adaptation in a later PR. We probably need to rename some variable names, etc.

Reply:
Let's just not forget about it.

Reply:
My objection is that arbitrary change of the variable names or unrelated request of the scope is bad practice. The reason is because usually you trigger side effects and it might cause a chain of reactions+additional checks,VR and etc. Second, you request to rename some variables, unrelated to the scope of this PR, which as for beginning they are not bad ones and the discussion should be in the case the PR have made changes on them. Which it doesnt and a very important reason of it is that there was a [PR](https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/12698) which reviewed and approved (requester was one of them) after a long talk on the rocket.chat if i remember. i just think is really bad to approve something and someone changed it without a good "reason". this causes more issues and inconsistency in the reviews and waste our time.

tespa. i would like to leave it as it is for the reasons i just mentioned. and you can decide later what its destiny will be :)

Reply:
*this change is ready for review*

Reply:
Another "tab or spaces" discussion :D

Reply:
> My objection is that arbitrary change of the variable names or unrelated request of the scope is bad practice. The reason is because usually you trigger side effects and it might cause a chain of reactions+additional checks,VR and etc. Second, you request to rename some variables, unrelated to the scope of this PR, which as for beginning they are not bad ones and the discussion should be in the case the PR have made changes on them. Which it doesnt and a very important reason of it is that there was a [PR](https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/12698) which reviewed and approved (requester was one of them) after a long talk on the rocket.chat if i remember. i just think is really bad to approve something and someone changed it without a good "reason". this causes more issues and inconsistency in the reviews and waste our time.
> 
> tespa. i would like to leave it as it is for the reasons i just mentioned. and you can decide later what its destiny will be :)

I agree with this, it's not something for this PR as I mentioned.